ta/arch/arm/link.mk

```diff
@@ -23,6 +23,14 @@ link-ldflags += -z max-page-size=4096 # OP-TEE always uses 4K alignment
 link-ldflags += --as-needed # Do not add dependency on unused shlib
 link-ldflags += $(link-ldflags$(sm))
 
+ifeq ($(CFG_TA_FTRACE_SUPPORT),y)
+dyn_list:
+	$(file > dyn_list,{__ftrace_info;};)
```

Comment:
Hmm, I think we can avoid the creation of file during evaluation via escaping `$` as follows:

```
--- a/ta/arch/arm/link.mk
+++ b/ta/arch/arm/link.mk
@@ -55,7 +55,7 @@ $(link-script-pp$(sm)): $(link-script$(sm)) $(conf-file) $(link-script-pp-makefi
 
 $(link-out-dir$(sm))/$(user-ta-uuid).elf: $(objs) $(libdeps) \
                                          $(link-script-pp$(sm)) \
-                                         $(ftracedep)
+                                         $$(ftracedep)
```

Your views?

Reply:
I don't think that will work. The easiest solution is probably to replace the `$(file ...)` stuff with `@echo ... > dyn_list`

Reply:
It seems `file` function could be used normally inside a rule (see [1]) which does operation during rule execution.

I thought your comment was regarding `$(eval $(call gen-link-t))` which causes evaluation before actual makefile execution and it could cause file creation during evaluation, so escaping `$` is one approach as suggested here [2].

[1] https://www.gnu.org/software/make/manual/html_node/File-Function.html#File-Function
[2] https://www.gnu.org/software/make/manual/html_node/Eval-Function.html#Eval-Function

Reply:
So maybe it works as expected, maybe it doesn't. I'd rather have that verified before we merge it.

By the way, I think `dyn_list` needs to be prefixed with `$(out-dir)` or maybe even the link out directory. Is this tested? I guess you would have noticed junk files in the git tree.

Reply:
...and even better with the proper trace/output macros ;-)

```make
$(link-out-dir$(sm))/dyn_list:
	@$(cmd-echo-silent) '  GEN     $@'
	$(q)echo "{__ftrace_info;};" >$@
```
(untested but should be right)

Reply:
Ok, let me use this instead.